js_modules/dagit/packages/core/src/launchpad/LaunchpadHooksContext.tsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+import {LaunchRootExecutionButton} from './LaunchRootExecutionButton';
+
+type LaunchpadHooksContextValue = {
+  LaunchRootExecutionButton?: typeof LaunchRootExecutionButton;
+};
+
+export const LaunchpadHooksContext = React.createContext<LaunchpadHooksContextValue>({
+  LaunchRootExecutionButton: undefined,
+});
+
+export function useLaunchPadHooks(): {
```

Comment:
This looks good to me! But I'm not super clear on what this hook does other than hide `useContext(LaunchpadHooksContext)`?  Might be nice future proofing!

Reply:
Yeah it just does `useContext` and returns either the original or an override if provided. I thought I  might need to reuse this logic since there other execution buttons so I put it in this hook